components/plan-build/bin/environment.sh

```diff
@@ -238,6 +238,54 @@ __populate_environment_from_deps() {
     fi
 }
 
+# For dependencies that do not have a RUNTIME_ENVIRONMENT file, we'll
+# at least assemble a PATH variable that can be used. To be complete
+# and utterly paranoid, we'll iterate through the TDEPS of the
+# dependency and assemble a PATH by layering all the PATHs of each of
+# those dependencies.
+#
+# Don't need to do this for packages that already have a
+# RUNTIME_ENVIRONMENT file, because this operation will have already
+# been done when those packages were built. This is to help preserve
+# backward compatibility.
+__assemble_legacy_path() {
+    local path_to_dep="${1}"
+    local path=""
+
+    warn "Assembling legacy path for ${path_to_dep}"
+    local tdeps=$(_read_metadata_file_for ${path_to_dep} "TDEPS")
+    if [ -n "${tdeps}" ]; then
```

Comment:
Not critical, but perhaps a Bash `[[` test?

Reply:
Yeah, that's probably a good call... I think I've just been habitually sticking to `[`, but I see a lot of your code uses `[[`, and we _are_ explicitly using `bash` and not `sh`.

Reply:
Out of curiosity, what's the benefit of `[[` over `[` here? I know some things require it, but if we're not using them, it seems like writing it the more portable way has merit.

Reply:
…and if we are explicitly using `bash` and not `sh`, maybe a `#!/bin/bash` would be helpful?

Reply:
@baumanj Probably consistency and keeping the mental overhead low. This code doesn't strictly require `[[`, but there are numerous other places in this package where we are explicitly relying on bash-only features, so portability across shells isn't really a concern (and if it were, there'd be a lot more to change than just the test procedures we use).